.github/workflows/pytest.yml

```diff
@@ -7,8 +7,11 @@ jobs:
   pytest:
     runs-on: ubuntu-latest
     steps:
-      - name: "Checkout metrics-utility (${{ github.ref }})"
+      - name: "Checkout metrics-utility (${{ github.event.pull_request.head.ref }})"
         uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
```

Comment:
I think merging this chunk, at a minimum, is required so that I can test the rest of the code.
@SherinV